Godeps/_workspace/src/k8s.io/kubernetes/pkg/securitycontextconstraints/capabilities/mustrunas_test.go

```diff
@@ -0,0 +1,369 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package capabilities
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	"reflect"
+	"testing"
+)
+
+func TestGenerateAdds(t *testing.T) {
+	tests := map[string]struct {
+		requiredCaps  *api.Capabilities
+		containerCaps *api.Capabilities
+		expectedCaps  *api.Capabilities
+	}{
+		"no required, no container requests": {
+			expectedCaps: nil,
+		},
+		"required, no container requests": {
+			requiredCaps: &api.Capabilities{
+				Add: []api.Capability{"foo"},
+			},
+			expectedCaps: &api.Capabilities{
+				Add: []api.Capability{"foo"},
+			},
+		},
+		"required, container requests add required": {
+			requiredCaps: &api.Capabilities{
+				Add: []api.Capability{"foo"},
+			},
+			containerCaps: &api.Capabilities{
+				Add: []api.Capability{"foo"},
+			},
+			expectedCaps: &api.Capabilities{
+				Add: []api.Capability{"foo"},
+			},
+		},
+		"multiple required, container requests add required": {
+			requiredCaps: &api.Capabilities{
+				Add: []api.Capability{"foo", "bar", "baz"},
+			},
+			containerCaps: &api.Capabilities{
+				Add: []api.Capability{"foo"},
+			},
+			expectedCaps: &api.Capabilities{
+				Add: []api.Capability{"bar", "baz", "foo"},
+			},
+		},
+		"required, container requests add non-required": {
+			requiredCaps: &api.Capabilities{
+				Add: []api.Capability{"foo"},
+			},
+			containerCaps: &api.Capabilities{
+				Add: []api.Capability{"bar"},
+			},
+			expectedCaps: &api.Capabilities{
+				Add: []api.Capability{"bar", "foo"},
+			},
+		},
```

Comment:
test case mismatches:
* container drop of "FOO" prevents an add of a requiredCapabilities.Add=[foo]
* container add of "FOO" is deduped with a requiredCapabilities.Add=[foo]
* container drop of "FOO" is deduped with a requiredCapabilities.Drop=[foo]

test deduping:
* requiredCapabilities.Add=[foo,foo,FOO] is deduped
* requiredCapabilities.Drop=[foo,foo,FOO] is deduped
* containerCapabilities.Add=[foo,foo,FOO] is deduped
* containerCapabilities.Drop=[foo,foo,FOO] is deduped